framework/src/relationshipmanagers/ProxyRelationshipManager.C

```diff
@@ -77,8 +77,10 @@ ProxyRelationshipManager::operator()(const MeshBase::const_element_iterator & /*
   // Build unique_id to elem map
   std::map<dof_id_type, const Elem *> unique_id_to_elem_map;
 
-  for (auto elem_it = _moose_mesh->getMesh().active_elements_begin();
-       elem_it != _moose_mesh->getMesh().active_elements_end();
+  // We have to use all elements because the ghosted elements from the other mesh may very well
+  // include inactive ancestors and things of that sort
+  for (auto elem_it = _moose_mesh->getMesh().elements_begin();
+       elem_it != _moose_mesh->getMesh().elements_end();
```

Comment:
For distributed mesh, you will get local+ghost, but for replicated mesh, you will have the whole world.  But I guess you only need local + ghost

Reply:
The thing is that here we are determining ghosting for the current mesh based on ghosting from the other mesh. Local + ghost for the current mesh may very well miss elements that the other mesh is going to tell us to ghost

Reply:
Oh, I see your point. Should we just have a clone somewhere so that one mesh is set and the other will be simple clone of the set one? It might be too late to clone ...

Reply:
Yes the original reason I had to make this change was because of adaptivity between time-steps which happens way after displaced mesh creation (where we do indeed just clone the reference mesh)

Reply:
It's back to https://github.com/libMesh/libmesh/issues/2661